analyses/0014-vitess/vitess-analysis.md

```diff
@@ -0,0 +1,1019 @@
+the right idea, but implementation must be improved if readers are to find and carry
+out the documented tasks.
+
+The User Guides require three types of changes to maximize their effectiveness:
+- Rename
+- Rewrite
+- Reorganize
+
+*Rename*
+
+Rename the individual pages in the User Guides to reflect the tasks that are described
+there. Use verbs ending in "-ing" (along with a noun that describes the object of
+the task, if applicable). This helps readers find the right content in at least two
+ways:
+- Makes the TOC more coherent
+- Gives meaningful results in the sitewide Search
+
+Here are three examples from the
+[same page]
+(https://vitess.io/docs/21.0/user-guides/configuration-basic/global-topo/)
+(the content of these sections is not considered here):
+- *Choosing a TopoRoot*: A good title. Describes the task ("choosing") and the object
+  of the task ("TopoRoot").
+- *Moving to a different TopoServer*: A good title. Describes the task ("moving").
+  Normally I'd recommend a more direct reference to the object --
+  "Moving TopoServers" -- but in this case the phrase ".. to a different TopoServer"
+  removes ambiguity.
+- *Backups*: Not a helpful title. It's not clear or what you're backing up or what
+  what aspect of the backups you're talking about. I'd change this to "Backing
+  up TopoServer data".
+
+Also on the Global TopoServer page, by the way:
+
+>  The following command line options are required for every Vitess component:
+>
+>  ```--topo_implementation=etcd2 --topo_global_server_address=<comma_separated_addresses>
+   --topo_global_root=/vitess/global```
+>
+>  To avoid repetition we will use <topo_flags> in our examples to signify the above
+   flags.
+
+Remove this. The "<topo_flags>" placeholder does not seem to have been implemented.
+There are no mentions of "<topo_flags"> elsewhere in the documentation, and in any
+case each would have to refer back to this page.
+
+*Rewrite*
+
+A reader typically consults the User Guide to find out how to do something. The User
+Guides should  consist primarily of procedures.
+
+There are some Vitess features for which the User Guide gives a description but does
+not provide adequate instructions for their use.
+
+For example, look at [Creating a Backup](https://vitess.io/docs/21.0/user-guides/operating-vitess/backup-and-restore/creating-a-backup/)
+
+This page appears to be well named ("Creating a Backup" is a descriptive task title)
+and properly placed (it resides in a logical location in the User Guide). However,
+the page itself doesn't contain an actual command or set of steps to back up a database:
+There is no actual command line given between *Configuration* and
+*Common Errors and Resolutions* The page's author might assume that it's implied,
+but it should be presented explicitly as a step in the procedure.
+
+Further down the page, another backup option, *Using mysqlshell*, has the same shortcomings:
+No actual command is presented.
```

Comment:
The commands are presented towards the end of the page https://vitess.io/docs/21.0/user-guides/operating-vitess/backup-and-restore/creating-a-backup/
Maybe the page needs to be re-organized.